File: z/数值的整数次方.cc
class Solution {
 public:
  double Power(double base, int exponent) {
    if (base == 0 && exponent == 0) {
      return 1;
    }
    if (base == 0) {
      return 0;
    }
    if (exponent == 0) {
      return 1;
    }
    double res = 1.0;
    int n = abs(exponent);
    for (int i = 0; i < n; ++i) {
      res *= base;
    }
    return exponent > 0 ? res : 1 / res;
  }
};

File: h/数值的整数次方.java
/*
给定一个double类型的浮点数base和int类型的整数exponent。求base的exponent次方。

保证base和exponent不同时为0
*/
public class Solution {
    public double Power(double base, int exponent) {
        double result = 1;
         
        if (exponent > 0) {
            for (int i = 1; i <= exponent; i++) {
               result *= base;
            }
        } else if (exponent < 0) {
            for (int i = 1; i <= -exponent; i++) {
               result *= base;
            }
            result = 1.0 / result;
        }
        return result;
  }
}
